tests/Summary/IsEmptyTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools\Tests\Summary;
+
+use IterTools\Summary;
+use IterTools\Tests\Fixture\DataProvider;
+use IterTools\Tests\Fixture\GeneratorFixture;
+use IterTools\Tests\Fixture\IteratorAggregateFixture;
+
+class IsEmptyTest extends \PHPUnit\Framework\TestCase
+{
+    use DataProvider;
+
+    /**
+     * @dataProvider dataProviderForEmptyIterable
+     * @param        iterable $iterable
+     */
+    public function testTrue(iterable $iterable)
+    {
+        // When
+        $result = Summary::isEmpty($iterable);
+
+        // Then
+        $this->assertTrue($result);
+    }
+
+    /**
+     * @dataProvider dataProviderForArrayFalse
+     * @param        array $iterable
+     */
+    public function testArrayFalse(array $iterable)
+    {
+        // When
+        $result = Summary::isEmpty($iterable);
+
+        // Then
+        $this->assertFalse($result);
+    }
+
+    public function dataProviderForArrayFalse(): array
+    {
+        return [
+            [[1]],
```

Comment:
Suggestion: I recommend adding some test cases for non-empty iterables that contain only "empty" values.

For example:
```php
[
    [0]
],
[
    [[]]
],
[
    [null]
],
```

Reply:
Done.